src/schema/v2/me/similarToRecentlyViewed.ts

```diff
@@ -19,16 +19,24 @@ export const SimilarToRecentlyViewed: GraphQLFieldConfig<
   resolve: async (
     { recently_viewed_artwork_ids },
     args,
-    { similarArtworksLoader }
+    { similarArtworksLoader, recentlyViewedArtworkIdsLoader, userID }
   ) => {
-    const recentlyViewedIds = recently_viewed_artwork_ids?.slice(0, 7) || []
-
+    if (!userID) return null
     const { page, size, offset } = convertConnectionArgsToGravityArgs(args)
 
     // Fetching all artworks until the current page because `offset` isn't working for similarArtworksLoader
     const numberOfArtworksToFetch = Math.min(size + offset, MAX_ARTWORKS)
 
     try {
+      let recentlyViewedIds = (recently_viewed_artwork_ids || [])?.slice(0, 7)
+
+      if (!!recentlyViewedIds) {
```

Comment:
With the line above this, `recentlyViewedIds` is always an array (either `[]` or with elements).

So this truth-y check is always returning `true` , and making an additional request (if it might not be needed) for the actual id's.

I think that logic is not quite right, this conditional should switch to: `!recentlyViewedIds.length` (if there were no id's present from the parent, _then_ fallback to fetching them explicitly.